### PR TITLE
Update and rename gsoc-project-proposal.md to gsoc-project-proposal.yml

### PR DESCRIPTION
**Pull request by MathewBiddle**
moving issue to issue form field template

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/gsoc-project-proposal.yml [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+      description: Include mentor names and GitHub handles as comma separated list:
+      placeholder: Mentor Name (@somegithubhandlethatsurelynoonealreadyhas), ...
+    validations:
+      required: false
+  - type: dropdown
+    id: size
+    attributes:
+      label: Expected Project Size:
+      options: 
+        - 175
```
Comment (ocefpaf) [inline .github/ISSUE_TEMPLATE/gsoc-project-proposal.yml:48]:
```suggestion
        - 90
        - 175
```

**Review by mwengren — APPROVED** [review]
Looks good!

**Comment by mwengren** [comment]
Do we need to update the link in the README as well here: https://github.com/ioos/gsoc/blob/main/README.md?plain=1#L12?

**Comment by MathewBiddle** [comment]
Thanks! @mwengren, yes! @ocefpaf recently recommended pointing folks to https://github.com/ioos/gsoc/issues/new/choose and then they can pick the template they want. I really like that approach as it reduced the url complexity and is a little more flexible.

**Comment by MathewBiddle** [comment]
FYI @mwengren I made the change to the readme in https://github.com/ioos/gsoc/commit/3e68e09009a01b0c06e9f175387a9d1d6e6b6c71

**Pull request merged**